Edit Product > Price Settings: support sale dates without a start or end date

jaclync opened the pull request:
Fixes #2020 
Fixes #2027 

## Changes

- Added a `UILabel` extension helper to apply body link styles
- In `ProductPriceSettingsViewModel`, updated the sale start/end date init and change handling based on the behavior in #2020, and updated unit tests
- In `ProductPriceSettingsViewController`:
  - If the sale end date is set, a "Remove end date" link cell is displayed that sets the end date to `nil`
  - The schedule sale is now visible whenever a sale start/end date is set (used to be only visible when both dates are set)

## Testing

Prerequisite: have 2 simple products in the store:
- one with a sale start date but no end date
- the other one with **a sale end date in the past** but no start date

### Missing end date

- Go to the Products tab
- Tap on the simple Product with a sale start date but no end date --> the price row should display the sale starting from a date
- Tap on the price settings row --> the end date should be empty
- Tap on the end date row and select a date --> the "Remove end date" row should appear
- Tap "Done" to go back to the edit product main screen
- Tap "Update" and wait for the modal to be gone --> the product should have both sale start and end dates set remotely

### Missing start date

- Go to the Products tab
- Tap on the simple Product with a sale end date but no start date --> the price row should display the sale ending on a date
- Tap on the price settings row --> the start date should be empty since the sale end date is in the past, and the "Remove end date" row should be visible
- Tap on the end date row and select a future date
- Tap "Done" to go back to the edit product main screen
- Tap on the price settings row --> the start date should show **today's date in the site time zone** since the sale end date is in the future, and the "Remove end date" row should be visible
- Tap "<" in the navigation bar --> it should go back to the edit product main screen without the discard changes action sheet
- Tap "Update" and wait for the modal to be gone --> the product should have just the end date updated to the future remotely

## Example screenshots

Screencast:
![simulator](https://user-images.githubusercontent.com/1945542/77496267-8cca5f00-6e85-11ea-90b5-f2ed7a9568f3.gif)

\ | edit product main screen | price settings
-- | -- | --
no start date | ![Simulator Screen Shot - iPhone 11 - 2020-03-25 at 10 27 29](https://user-images.githubusercontent.com/1945542/77496250-85a35100-6e85-11ea-816e-96dee15c2198.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-25 at 10 27 30](https://user-images.githubusercontent.com/1945542/77496261-8a680500-6e85-11ea-9089-ca26fbaeb59f.png)
no end date | ![Simulator Screen Shot - iPhone 11 - 2020-03-25 at 10 27 36](https://user-images.githubusercontent.com/1945542/77496263-8b009b80-6e85-11ea-8e6d-050fe44e135b.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-25 at 10 27 38](https://user-images.githubusercontent.com/1945542/77496264-8b993200-6e85-11ea-81c6-462e2eb1744c.png)




Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Garance91540 commented:
@jaclync it feels a bit weird to have the "To" row empty https://d.pr/i/sB7kYz. Is it possible to add placeholders "Select start date" / "Select end date" when the fields are expanded but empty?

jaclync commented:
@Garance91540 yea it does look odd when the value is empty, I added the placeholder and it looks like:

missing sale end date | missing sale start date (only when the sale end date is in the past)
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-03-26 at 10 29 46](https://user-images.githubusercontent.com/1945542/77604733-2a896100-6f4e-11ea-9927-2847593e0144.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-26 at 10 33 29](https://user-images.githubusercontent.com/1945542/77604660-075eb180-6f4e-11ea-818b-8ff981ca7c66.png)

@pmusolino I also updated the PR (release notes and adding placeholder) in the 2 commits above!

pmusolino commented:
Looks good to me!
